CMakeLists.txt

```diff
@@ -141,6 +144,18 @@ set(CMAKE_MODULE_PATH
 )
 
 
+# Set linker and compiler flags based on arguments.
+if (VIAMCPPSDK_USE_WALL_WERROR)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror")
```

Comment:
Applying these to `CMAKE_CXX_FLAGS` may be too big a hammer. I'd suggest looking into whether you can achieve this with `add_compile_options` and `add_link_options` at the directory level:
- https://cmake.org/cmake/help/latest/command/add_compile_options.html
- https://cmake.org/cmake/help/latest/command/add_link_options.html
You probably also want to apply these as `PRIVATE` so they don't propagate to SDK consumers that use the library. I'm actually not sure if you can do that with `add_compile_options` / `add_link_options`. You could do it with `target_compile_options` for each target but that's a pain to need to add everywhere. Hopefully there is a way to both make it private and apply it once.

Reply:
Some interesting thoughts on doing this here: https://www.foonathan.net/2018/10/cmake-warnings/

Reply:
Ack, appreciate the feedback! Will look into this and push updates.